### PR TITLE
feat: adding pull request template

### PR DESCRIPTION
**Pull request by abanuelo**
The goal of this PR is to introduce a way to format PR's by allowing users to:

1. Descriptions
2. Link back to existing issue that trickled the need for the PR
3. PR Checklist satisfying that formatting, testing, and relevant documentation
4. Additional info (ie Jira ticket)

### CODE REVIEW

File: .github/workflows/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+### Description
+<!-- Provide a brief description of the changes introduced by this pull request -->
+
+### Issue Link
+<!-- Provide a link to the related issue on GitHub or another issue tracking system -->
+
+### Checklist
+- [ ] I have tested the changes locally via `pytest` and/or other means
+- [ ] I have added or updated relevant documentation
+- [ ] I have auto formatted the code to black and istio conventions
```
Comment (dfremont) [inline .github/workflows/PULL_REQUEST_TEMPLATE.md:10]:
```suggestion
- [ ] I have autoformatted the code with black and isort
```

**Review by dfremont — CHANGES_REQUESTED** [review]
Looks great, just one tiny comment above.

File: .github/workflows/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (abanuelo) [reply .github/workflows/PULL_REQUEST_TEMPLATE.md]: Ack!

**Review by Eric-Vin — APPROVED** [review]
Looks good to me!

**Comment by dfremont** [comment]
SCENIC-12

**Comment by dfremont** [comment]
[SCENIC-12]

**Review by dfremont — APPROVED** [review]

**Pull request merged**